server/src/internalClusterTest/java/org/opensearch/search/backpressure/SearchBackpressureIT.java

```diff
@@ -0,0 +1,269 @@
+import org.opensearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.SUITE)
+public class SearchBackpressureIT extends OpenSearchIntegTestCase {
+
+    private static final TimeValue TIMEOUT = new TimeValue(30, TimeUnit.SECONDS);
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        final List<Class<? extends Plugin>> plugins = new ArrayList<>(super.nodePlugins());
+        plugins.add(TestPlugin.class);
+        return plugins;
+    }
+
+    @Before
+    public final void setupNodeSettings() {
+        Settings request = Settings.builder()
+            .put(NodeDuressSettings.SETTING_CPU_THRESHOLD.getKey(), 0.0)
+            .put(NodeDuressSettings.SETTING_HEAP_THRESHOLD.getKey(), 0.0)
+            .put(NodeDuressSettings.SETTING_NUM_SUCCESSIVE_BREACHES.getKey(), 1)
+            .put(SearchShardTaskSettings.SETTING_TOTAL_HEAP_PERCENT_THRESHOLD.getKey(), 0.0)
+            .build();
+        assertAcked(client().admin().cluster().prepareUpdateSettings().setPersistentSettings(request).get());
+    }
+
+    @After
+    public final void cleanupNodeSettings() {
+        assertAcked(
+            client().admin()
+                .cluster()
+                .prepareUpdateSettings()
+                .setPersistentSettings(Settings.builder().putNull("*"))
+                .setTransientSettings(Settings.builder().putNull("*"))
+        );
+    }
+
+    public void testSearchShardTaskCancellationWithHighElapsedTime() throws InterruptedException {
+        Settings request = Settings.builder()
+            .put(SearchBackpressureSettings.SETTING_MODE.getKey(), "enforced")
+            .put(ElapsedTimeTracker.SETTING_ELAPSED_TIME_MILLIS_THRESHOLD.getKey(), 5000)
+            .build();
+        assertAcked(client().admin().cluster().prepareUpdateSettings().setPersistentSettings(request).get());
+
+        CountDownLatch latch = new CountDownLatch(1);
+
+        client().execute(TestTransportAction.ACTION, new TestRequest(RequestType.HIGH_ELAPSED_TIME), new ActionListener<>() {
+            @Override
+            public void onResponse(TestResponse testResponse) {
+                fail("SearchShardTask should have been cancelled");
+                latch.countDown();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                assertEquals(TaskCancelledException.class, e.getClass());
+                assertTrue(e.getMessage().contains("elapsed time exceeded"));
+                latch.countDown();
+            }
+        });
+
+        latch.await(TIMEOUT.getSeconds(), TimeUnit.SECONDS);
```

Comment:
You should assert that all these `await` calls return true.

Reply:
We are having the same timeout value for the request execution as well, and if the TestRequest times out, it will make a call to `onResponse()` and we already have necessary assertions there.

Reply:
I realize it is fine now but this is more about maintainability. Future changes may violate some of the current assumptions so it is better to explicitly verify that things are working as expected.